### ISSUE TITLE
Enable CORS for OAuth endpoints

### ISSUE THREAD

**Issue by philippwaller** [openhab/openhab-core#1936]
The OAuth endpoints are currently protected against cross-origin resource sharing (CORS) by default. Analogous to the REST API, it should be possible to activate CORS via the parameter `org.openhab.cors:enable=true`. Currently this is not possible. Therefore, external web clients can only use the API to a limited extent.

**Comment by ghys** [comment]
after seeing in the logs:
```
[INFO ] [re.io.rest.internal.filter.CorsFilter] - enabled CORS for REST API.
```
```
curl -v -H 'Origin: https://example.com' http://localhost:8080/rest/auth/token
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Expire in 200 ms for 4 (transfer 0x7ffff034f140)
* Connected to localhost (127.0.0.1) port 8080 (#0)
> GET /rest/auth/token HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.64.0
> Accept: */*
> Origin: https://example.com
>
< HTTP/1.1 405 Method Not Allowed
< Allow: DELETE,POST,GET,PUT,OPTIONS,HEAD
< Date: Tue, 15 Dec 2020 08:31:30 GMT
< Access-Control-Allow-Origin: https://example.com
< Content-Length: 0
< Server: Jetty(9.4.20.v20190813)
```
I get the `Access-Control-Allow-Origin: https://example.com` header.

**Comment by philippwaller** [comment]
Sorry, I left out some details. I was taking about the authorization endpoint `http://localhost:8080/auth`.
```
curl -v -H 'Origin: https://example.com' http://localhost:8080/auth/.well-known/openid-configuration
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 8080 (#0)
> GET /auth/.well-known/openid-configuration HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.64.1
> Accept: */*
> Origin: https://example.com
>
< HTTP/1.1 200 OK
< Content-Type: text/plain;charset=utf-8
< Content-Length: 15
< Server: Jetty(9.4.20.v20190813)
<
* Connection #0 to host localhost left intact
invalid_request* Closing connection 0
```

However, I just noticed that the OpenID Connect Discovery is not implemented yet.  Therefore, there is probably no reason to access the endpoint `/auth` via CORS requests.

Thank you Yannick @ghys!

**Comment by ghys** [comment]
That's right, none of OIDC is implemented currently, would be nice to have one day but for now it's only a simplified OAuth2 authorization code flow (with no pre-registered clients but some checks about the clientId against the callback URL instead).

**Comment by philippwaller** [comment]
I have found another issue that is directly related. CORS requests against protected endpoints are failing with the following error:
    
    Access to XMLHttpRequest at 'http://localhost:8080/rest/items' from origin 'http://localhost:4200' 
    has been blocked by CORS policy: Request header field authorization is not allowed by 
    Access-Control-Allow-Headers in preflight response.

If I capture the preflight `OPTIONS` request with Firefox dev tools, I can see that the authorization header is not listed in the `Access-Control-Allow-Headers`.

![image](https://user-images.githubusercontent.com/1090452/102254525-01a86200-3f09-11eb-91f0-e8e76cc9d287.png)

**Comment by philippwaller** [comment]
For the record: I created a Pull Request (#1963) to address this issue.
